src/components/twilio-voice-basic-call-control/public/twilio-voice-basic-call-control.js

```diff
@@ -22,75 +19,165 @@ class TwilioVoiceBasicCallControl extends HTMLElement {
-      this.#showButtons('#hold-buttons', 'hold');
-      this.#showButtons('#mute-buttons', 'mute');
+      this.#showElement('#add-participant-container', true);
+      this.#renderCallControlButtons();
     }
   }
 
-  async #handleHold(shouldHold) {
-    if (this.#hasParticipantConnected()) {
-      const response = await this.#updateConferenceCall({ hold: shouldHold });
-      if (response.status === 200) {
-        this.#showButtons('#hold-buttons', shouldHold ? 'resume' : 'hold');
-      } else {
-        console.error('Unable to set hold: ', response.error);
-      }
+  async #handleHold(shouldHold, callSid) {
+    const response = await this.#updateConferenceCall(callSid, {
+      hold: shouldHold,
+    });
+    if (response.status === 200) {
+      this.#showElement(`#${callSid}-hold`, !shouldHold);
+      this.#showElement(`#${callSid}-resume`, shouldHold);
+    } else {
+      console.error('Unable to set hold: ', response.error);
     }
   }
 
-  async #handleMute(shouldMute) {
-    if (this.#hasParticipantConnected()) {
-      const response = await this.#updateConferenceCall({ muted: shouldMute });
-      if (response.status === 200) {
-        this.#showButtons('#mute-buttons', shouldMute ? 'unmute' : 'mute');
-      } else {
-        console.error('Unable to set mute: ', response.error);
-      }
+  async #handleMute(shouldMute, callSid) {
+    const response = await this.#updateConferenceCall(callSid, {
+      muted: shouldMute,
+    });
+    if (response.status === 200) {
+      this.#showElement(`#${callSid}-mute`, !shouldMute);
+      this.#showElement(`#${callSid}-unmute`, shouldMute);
+    } else {
+      console.error('Unable to set mute: ', response.error);
     }
   }
 
-  #hasParticipantConnected = () =>
-    Boolean(this.#call && this.#callSid && this.#conferenceSid);
+  #removeCallControlButtons() {
+    const callControlButtons = this.shadowRoot.querySelector(
+      '#call-control-buttons'
+    );
+    if (callControlButtons) {
+      callControlButtons.remove();
+    }
+  }
 
   #render() {
     this.shadowRoot.innerHTML = `
-      <div>
-        <div id="hold-buttons">
-          <button id="hold">Hold</button>
-          <button id="resume">Resume</button>
-        </div>
-        <div id="mute-buttons">
-          <button id="mute">Mute</button>
-          <button id="unmute">Unmute</button>
-        </div>
+      <div id="basic-call-control">
+        <div id="add-participant-container" style="display: none;">
+          <input
+            type="text"
+            placeholder="Add participant"
+            id="add-participant"
+            value=""
+          />
+          <button id="add-conference">Add</button>
+        <div>
       </div>
     `;
   }
 
+  #renderCallControlButton(options) {
+    const { name, callSid, shouldHide } = options;
+    const button = document.createElement('button');
+    button.innerHTML = name;
+    button.setAttribute('id', `${callSid}-${name}`);
```

Comment:
I'm not sure it's common practice to set the ID of an HTML element like this since IDs are tightly tied to CSS. Maybe some other attribute we can change? Data attributes, or do we want to use a custom element that has a CallSID attribute?

Reply:
I understand the concern, but I don't think setting the `id` when an element is created should cause us problems. I think adding the `callSid` to the `id` pretty much guarantees us a unique `id` for every button.